### DISCUSSION TITLE
How can a fork join commonhaus?

### DISCUSSION THREAD

**Discussion by velo** [commonhaus/foundation#170] [Joining Commonhaus]
### Project information

This is more a curve ball than other projects.

A while back I forked querydsl and got it alive and kicking.  Current developers are neither interested in continuing querydsl, nor handing over, nor accepting help.

https://github.com/OpenFeign/querydsl/

In the past 3 years, the only activity you will see is a single torpedo release around the time my fork made enough noise.

I would love to bring that code into commonhaus too, but I double that would be straight forward.

- **Project name**: Querydsl? OpenDSL? OpenQuery? GeneralQuery?
- **Project website**:  https://github.com/OpenFeign/querydsl/
- **Code repository**: https://github.com/OpenFeign/querydsl/
- **License**: Apache License 2

### Do you have authority to represent this project?

- [ ] I own or maintain this project

### Describe your project

Querydsl is a framework which enables the construction of type-safe SQL-like queries for multiple backends including JPA, MongoDB and SQL in Java.

Instead of writing queries as inline strings or externalizing them into XML files they are constructed via a fluent API.

### Things that must be done (bootstrap phase)

- [ ] **Review the bylaws**, especially sections about [projects](https://www.commonhaus.org/bylaws/cf-membership.html#projects), [decision making](https://www.commonhaus.org/bylaws/decision-making.html), and [voting for significant decisions](https://www.commonhaus.org/bylaws/cf-council.html#voting).
- [ ] **Read the [bootstrapping agreement](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md)**.
- [ ] If all is well, and you agree to participate in the bootstrapping phase to help us build this thing, then **create a PR** with the following:

    1. A filled out copy of the bootstrapping agreement. You can make a copy of [bootstrapping.md](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md) file and check that in, or checkin a filled out [pdf](https://github.com/commonhaus/foundation/releases/download/final-draft/bootstrapping.pdf). Please follow naming conventions.
    2. An updated [CONTACTS.yaml](https://github.com/commonhaus/foundation/blob/main/CONTACTS.yaml) that names your project representative in the EGC section
    3. An updated [PROJECTS.yaml](https://github.com/commonhaus/foundation/blob/main/PROJECTS.yaml) that provides details / describes your project
    4. An updated [TRADEMARKS.md](https://github.com/commonhaus/foundation/blob/main/TRADEMARKS.md) that adds your project name and project logo (if you have one). Please include the footnote reference, as we are using your project name and logo with your permission until we have a formal agreement in place and resources are transferred.

    The commit should be "Signed-off-by" your GH login (`git commit -s -m "<project> bootstrapping agreement, yyyy-mm-dd"`)

**Comment by maxandersen** [comment]
Trying to grok more - is https://github.com/OpenFeign/querydsl/ the original or the fork? if the fork where is the original?

**Reply by velo** [reply]
https://github.com/OpenFeign/querydsl/ is the fork
 
https://github.com/querydsl/querydsl/ is the original moribund project

**Reply by ebullient** [reply]
It isn't _completely_ moribund.. only mostly moribund, which makes it harder.

**Reply by velo** [reply]
Na, it's completely moribund.

There was a single torphaedo release and that was it. Not even release notes

**Reply by velo** [reply]
Would rebranding it solve the issue? 

CommonDSL?

Can publish old querydsl-* modules with API compatible classes

**Reply by ebullient** [reply]
Yes, I think that would help.. it would also make it easier to describe. "Which version of QueryDSL" becomes a hard question to answer if they keep the same name (and that's without considering Copyright or Trademark issues)

**Reply by evanchooly** [reply]
If we're rebranding it, might I suggest `QueryDiesel`?

**Comment by maxandersen** [comment]
@velo is there any trace of response from the original founders on why they aren't interested in handing over?

**Reply by velo** [reply]
When fork start getting some traction they got active for a day or so, said something about having moved on, would give more people access to project things, vanished for a week, said couldn't take my changes in as they were not willing to maintain code that they didn't reviewed (which was kinda rich, since they are not maintaining current code basis either), did the torpedo release and vanished again.

IMHO, feels like they don't want to do it, but don't want anyone else do take over either.

Original author moved on for years now, I tried to reach out, but no joy.

**Reply by maxandersen** [reply]
okey - so it is always a delicate matter when a project in such a situation. We can't really know for sure if the original authors are okey or not by it being forked.

In general I would suggest to at least consider do one final outreach to get ack that it is okey to use the name/move the project; if not or no reply you would have do to a significant enough rename/rebranding to make sure there is no confusion.

**Reply by gavinking** [reply]
> We can't really know for sure if the original authors are okey or not by it being forked.

By definition, when one releases open source code, one is explicitly making it OK to fork.

**Reply by ebullient** [reply]
right... but making it ok to fork != making it ok to keep the same name

**Reply by gavinking** [reply]
Sure.

**Reply by maxandersen** [reply]
yes, by fork here means keeping name which is the problematic part.

**Comment by velo** [comment]
Will keep as is, forked without any relationship with commonhaus for the next year or so.
